libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt

```diff
@@ -151,19 +155,29 @@ class NavigationRoute internal constructor(
             routerOrigin: RouterOrigin,
             routeParser: SDKRouteParser = NativeRouteParserWrapper
         ): List<NavigationRoute> {
+            logI("NavigationRoute.create called for $routeRequestUrl", TAG)
             return coroutineScope {
                 val deferredResponseParsing = async(ThreadController.DefaultDispatcher) {
-                    DirectionsResponse.fromJson(directionsResponseJson)
+                    logI("parsing directions response to java model from $routeRequestUrl", TAG)
+                    val result = DirectionsResponse.fromJson(directionsResponseJson)
+                    logI("parsed directions response to java model from $routeRequestUrl", TAG)
+                    result
                 }
                 val deferredNativeParsing = async(ThreadController.DefaultDispatcher) {
-                    routeParser.parseDirectionsResponse(
+                    logI("parsing directions response to RouteInterface from $routeRequestUrl", TAG)
+                    val result = routeParser.parseDirectionsResponse(
                         directionsResponseJson,
                         routeRequestUrl,
                         routerOrigin,
                     )
+                    logI("parsed directions response to RouteInterface from $routeRequestUrl", TAG)
+                    result
                 }
                 val deferredRouteOptionsParsing = async(ThreadController.DefaultDispatcher) {
-                    RouteOptions.fromUrl(URL(routeRequestUrl))
+                    logI("parsing request url to RouteOptions: $routeRequestUrl", TAG)
+                    val result = RouteOptions.fromUrl(URL(routeRequestUrl))
+                    logI("parsed request url to RouteOptions: $routeRequestUrl", TAG)
+                    result
```

Comment:
looks overkill a bit
```suggestion
                    RouteOptions.fromUrl(URL(routeRequestUrl)).also { routOptions ->
                      logI(" Request URL has been parsed to RouteOptions: [$routeRequestUrl] to [${routOptions}]", TAG)
                    }
                    
                    result
```

sorry, for the brackets :)

Reply:
partially applied, thanks 👍